controllers/infrastructure.cluster.x-k8s.io/kopsmachinepool_controller.go

```diff
@@ -166,9 +154,15 @@ func (r *KopsMachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return resultDefault, nil
 	}
 
-	igName, err := getInstanceGroupNameFromKopsMachinePool(kopsMachinePool)
-	if err != nil {
-		return resultError, err
+	igName := kopsMachinePool.Name
+
+	if _, ok := kopsMachinePool.Spec.KopsInstanceGroupSpec.NodeLabels["kops.k8s.io/instance-group-name"]; !ok {
```

Comment:
If we are going to replace the label in the next line, why do we need to verify if this is defined? Do we use this for anything else? Should we wrap this in a validation function at the beginning of the reconciliation?

Reply:
Removed the verification, now it only checks if there's any nodelabels defined and add the instance-group-name tag.